better error message when jdk not installed

Was reinstalling `ixmp` on a fresh system and got the following error due to a `jdk` flavor not being installed:

```
________________________________________________________________________________________ ERROR at setup of test_scen_list _________________________________________________________________________________________

request = <SubRequest 'test_mp' for <Function test_scen_list>>
tmp_env = environ({'CLUTTER_IM_MODULE': 'xim', 'CONDA_SHLVL': '2', 'PYTHONUNBUFFERED': 'nonempty string', 'LD_LIBRARY_PATH': '/h...', 'PYTEST_CURRENT_TEST': 'tests/test_core.py::test_scen_list (setup)', 'IXMP_DATA': '/tmp/pytest-of-gidden/pytest-2'})
test_data_path = PosixPath('/home/gidden/work/iiasa/message/ixmp/tests/data')

    @pytest.fixture(scope='function')
    def test_mp(request, tmp_env, test_data_path):
        """An ixmp.Platform connected to a temporary, local database."""
>       yield from create_test_mp(request, test_data_path, 'ixmptest')

../../../../.local/lib/python3.7/site-packages/ixmp-0.2.0.post0.dev286-py3.7.egg/ixmp/testing.py:87: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../../.local/lib/python3.7/site-packages/ixmp-0.2.0.post0.dev286-py3.7.egg/ixmp/testing.py:108: in create_test_mp
    mp = Platform(name=platform_name)
../../../../.local/lib/python3.7/site-packages/ixmp-0.2.0.post0.dev286-py3.7.egg/ixmp/core.py:122: in __init__
    self._backend = backend_class(**kwargs)
../../../../.local/lib/python3.7/site-packages/ixmp-0.2.0.post0.dev286-py3.7.egg/ixmp/backend/jdbc.py:193: in __init__
    start_jvm(jvmargs)
../../../../.local/lib/python3.7/site-packages/ixmp-0.2.0.post0.dev286-py3.7.egg/ixmp/backend/jdbc.py:761: in start_jvm
    args = [jpype.getDefaultJVMPath()]
../../../../.local/lib/python3.7/site-packages/JPype1-0.7.0-py3.7-linux-x86_64.egg/jpype/_core.py:337: in getDefaultJVMPath
    return finder.get_jvm_path()
../../../../.local/lib/python3.7/site-packages/JPype1-0.7.0-py3.7-linux-x86_64.egg/jpype/_jvmfinder.py:160: in get_jvm_path
    jvm = method()
../../../../.local/lib/python3.7/site-packages/JPype1-0.7.0-py3.7-linux-x86_64.egg/jpype/_jvmfinder.py:215: in _get_from_known_locations
    for home in self.find_possible_homes(self._locations):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <jpype._linux.LinuxJVMFinder object at 0x7f63319bdcf8>, parents = ('/usr/lib/jvm', '/usr/java', '/opt/sun')

    def find_possible_homes(self, parents):
        """
        Generator that looks for the first-level children folders that could be
        Java installations, according to their name
    
        Parameters:
            parents (str[]): A list of parent directories
    
        Returns:
            A list of the possible JVM installation folders
        """
        homes = []
        java_names = ('jre', 'jdk', 'java')
    
        for parent in parents:
>           for childname in sorted(os.listdir(parent)):
E           FileNotFoundError: [Errno 2] No such file or directory: '/usr/lib/jvm'

../../../../.local/lib/python3.7/site-packages/JPype1-0.7.0-py3.7-linux-x86_64.egg/jpype/_jvmfinder.py:120: FileNotFoundError
```

Perhaps it is worth catching this `FileNotFoundError` and suggesting the user install the missing package?

--- a/ixmp/backend/jdbc.py
+++ b/ixmp/backend/jdbc.py
@@ -1114,7 +1114,15 @@ def start_jvm(jvmargs=None):
     log.debug(f"jpype.getDefaultJVMPath: {jpype.getDefaultJVMPath()}")
     log.debug(f"args to startJVM: {args} {kwargs}")
 
-    jpype.startJVM(*args, **kwargs)
+    try:
+        jpype.startJVM(*args, **kwargs)
+    except FileNotFoundError as e:  # pragma: no cover
+        # Not covered by tests. jpype.getDefaultJVMPath() tries an extensive set of
+        # methods to find the JVM; it would require excessive effort to defeat these.
+        raise FileNotFoundError(
+            "This error may occur because you have not installed or configured a Java"
+            "Runtime Environment. See the install documentation."
+        ) from e
 
     # define auxiliary references to Java classes
     global java

